src/main.rs

```diff
@@ -92,29 +92,71 @@ impl HanoiError {
 ///    * `Action`: if the input was well formed
 ///    * `Hanoi::UnknownCommand`: otherwise
 fn parse_action(input: &str) -> Result<Action,HanoiError> {
-    unimplemented!()
+    let mut iter  = input.chars();
+    let first  = iter.next();
+    let second = iter.next();
+    let third  = iter.next();
+
+
+    if (first, second, third) == (Some('q'), None, None) {
+        return Ok(Action::Quit)
+    }
+
+    match (char_to_peg(first), char_to_peg(second), char_to_peg(third)) {
+        (Some(fromPeg), Some(toPeg), None) => return Ok(Action::Move(Move::new(fromPeg, toPeg))),
+        _ => return Err(HanoiError::UnknownCommand),
+    }
+}
+
+fn char_to_peg(ch: Option<char>) -> Option<Peg> {
+    match ch {
+        Some('l') => Some(Peg::Left),
+        Some('c') => Some(Peg::Center),
+        Some('r') => Some(Peg::Right),
+        _ => None,
+    }
 }
 
 impl State {
 
     /// Creates a Towers of Hanoi game with `disks` disks in a single tower
     fn new(disks: u8) -> State {
-        unimplemented!()
+        let mut left = Vec::new();
+        let center = Vec::new();
+        let right = Vec::new();
+
+        for i in 0..disks {
+            left.insert(i as usize, Disk(disks - i));
+        }
```

Comment:
This works, but you should also look into `Iterator::map` and `Iterator::collect`